### PR TITLE
Feature/movie fetch

### LINKED ISSUE #6: 3.1 Implement Movies API Endpoint
Movies Endpoint - 
User Story - As a user, I want to view all movies that are in the api, so that I can select a movie for more details

Acceptance Criteria - 
Scenario - The user loads the webpage and all movie posters from the api are clearly displayed

### PR DESCRIPTION
**Pull request by williamphelps13**
# Thank you for pushing your code!

## What was implemented?
- Add fetch for single movie
- Small styling adjustments for single movie page
- Closes #6 

## Were there any issues that arose?
- The fetch was running before the poster click
- We overcame this by putting the fetch in a method that was called on the click

## Is there anything that you need from your teammate?
- Next step will be to implement error handling for 500 errors 

## Any other comments, questions, or concerns?
- We also want to circle back and add a background image to the movie page and round the movie rating

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?
- [ ] open index.html
- [ ] console.log()
- [x] dev tools

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 .App-header {
   background-color: #282c34;
-  min-height: 7rem;
+  min-height: 15vh;
```
Comment (hayleyw7) [inline src/App.css:7]: This make it look so much better.

File: src/App.js [hunk 2]
```diff
@@ -9,34 +8,39 @@ class App extends Component {
     super();
     this.state = {
       movies: [],
-      singleMovie: movieData.singleMovie,
+      singleMovie: {},
```
Comment (hayleyw7) [inline src/App.js:11]: We got this updating the empty object.

File: src/App.js [hunk 3]
```diff
@@ -9,34 +8,39 @@ class App extends Component {
...
     fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
       .then(response => response.json())
       .then(data => {this.setState({movies: data.movies})})
       .catch(error => this.setState({error: error}))
   }
-
-  clickPoster = () => {
+  
+  clickPoster = (id) => {   
+    this.fetchMovie(id);
```
Comment (hayleyw7) [inline src/App.js:24]: This fixed the order of operations. Hooray!

File: src/Movie.css [hunk 4]
```diff
@@ -1,3 +1,10 @@
+.movie-background {
+  display: flex;
+  justify-content: center;
```
Comment (hayleyw7) [inline src/Movie.css:3]: I love the way we got this set up!

File: src/Posters.js [hunk 5]
```diff
@@ -9,8 +9,8 @@ const Posters = ({ posters, clickPoster }) => {
         type="image"
         src={poster.poster_path}
         className='poster-icon'
-        onClick={() => clickPoster()}
-        alt='(movie title)'
+        onClick={() => clickPoster(poster.id)}
+        alt={poster.title}
```
Comment (hayleyw7) [inline src/Posters.js:13]: Accessibility is important, yes!

File: src/App.js [hunk 6]
```diff
@@ -9,34 +8,39 @@ class App extends Component {
...
     document.querySelector('.posters-container').classList.add('hidden');
-    document.querySelector('.movie-background').style.background = `url(${this.state.singleMovie.backdrop_path})`;
```
Comment (hayleyw7) [inline src/App.js:26]: We should be able to add something like this back in later if we'd like to do so.

**Review by hayleyw7 — APPROVED** [review]
Great PR! Thanks for working together on this! More detailed code review in notes.

File: src/App.js [hunk 7]
```diff
@@ -9,34 +8,39 @@ class App extends Component {
...
       movies: [],
-      singleMovie: movieData.singleMovie,
+      singleMovie: {},
       error: ''
     }
   }
-
+  
   componentDidMount() {
     fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
```
Comment (sertmer) [inline src/App.js:17]: don't forget to define the network requests in their own file, per the rubric 😄

File: src/App.js [hunk 8]
```diff
@@ -9,34 +8,39 @@ class App extends Component {
...
+
+  fetchMovie(id) {
+    return fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${id}`)
+      .then(response => response.json())
+      .then(data => {this.setState({singleMovie: data.movie})})
+      .catch(error => this.setState({error: error}))
   }
 
   clickBackBtn = () => {
     document.querySelector('.posters-container').classList.remove('hidden');
```
Comment (sertmer) [inline src/App.js:36]: Hopefully you'll be able to refactor these query selectors out once you refactor for Router 👍

File: src/Posters.js (hunk 5 above)
Comment (sertmer) [reply src/Posters.js:13]: 👏

File: src/App.js (hunk 2 above)
Comment (sertmer) [reply src/App.js:11]: An empty object totally works! You could also reach for Null if you wanted to!

**Review by sertmer — COMMENTED** [review]
Good stuff! Check my comments for some refactoring ideas

**Pull request merged**